### PR TITLE
feat: configurable chunk size

### LINKED ISSUE #112: Configurable chunk size
# Feature Request

Currently, the chunk size in the data sink is hard-coded to 500MB as CHUNK_SIZE_IN_BYTES. This could limit users in particularl environments with constrained memory resources.
We propose making the chunk size configurable via the EDC configuration.

## Which Areas Would Be Affected?
S3-Dataplane-extension , S3-Datasink

## Why Is the Feature Desired?
When edc is being deployed in an environment which has limited memory size 500mb chunk size is too much and dataplane cannot transfer a large size file

## Solution Proposal
Get the value from configuration , and after validation pass it to the datasink.

### PR DESCRIPTION
**Pull request by git-masoud**
## What this PR changes/adds

Chunk size can be configured via the EDC configuration.

## Why it does that

Currently, the chunk size in the data sink is hard coded to 500MB as CHUNK_SIZE_IN_BYTES. This could limit users in particular environments with constrained memory resources.

## Linked Issue(s)

Closes #112

### CODE REVIEW

File: extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/DataPlaneS3Extension.java [hunk 1]
```diff
@@ -29,6 +30,12 @@
 public class DataPlaneS3Extension implements ServiceExtension {
 
     public static final String NAME = "Data Plane S3 Storage";
+
+    static final int DEFAULT_CHUNK_SIZE_IN_MB = 500; // 500MB chunk size
+
+    @Setting
```
Comment (ndr-brt) [inline extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/DataPlaneS3Extension.java:36]: please add details to the `@Setting` annotation, like `value`, `defaultValue`, `type` and so on

File: extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java [hunk 2]
```diff
@@ -140,6 +141,10 @@ public Builder chunkSizeBytes(int chunkSize) {
         }
 
         @Override
-        protected void validate() {}
+        protected void validate() {
+            if (this.sink.chunkSize < 1) {
+                throw new IllegalArgumentException("Chunk size must be greater than zero! Actual value is: " + this.sink.chunkSize);
+            }
```
Comment (ndr-brt) [inline extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java:147]: please avoid this lazy validation, if the chunkSize is not configured correctly, the dataplane should not start at all, please validate it in the `initialize` method of the extension.

**Review by ndr-brt — CHANGES_REQUESTED** [review]

File: extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java (hunk 2 above)
Comment (git-masoud) [reply extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java:147]: I've added it into the **_initialize_** method of the extension and left it here as well.

Comment (ndr-brt) [reply extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java:147]: here does not make any sense anymore, please remove it

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**